File: Haskell/8 kyu/8 kyu - Calculate average.hs
-- https://www.codewars.com/kata/calculate-average/train/haskell
-- My solution
module Average where

    avg :: [Float] -> Float
    avg l = (sum l) / fromIntegral (length l)
-- ...
module Average where

    import Data.List (genericLength)

    avg :: [Float] -> Float
    avg l = sum l / genericLength l
-- ...
module Average where

    avg :: [Float] -> Float
    avg [] = error "Cannot get average of empty list of values."
    avg l = (sum l) / (fromIntegral $ length l)
-- ...
module Average where

    avg :: [Float] -> Float
    avg  = pure (/) <*> sum <*> fromIntegral . length
-- ...
module Average where

    import Control.Monad (liftM2)
    import Data.List (genericLength)
    
    avg :: [Float] -> Float
    avg = liftM2 (/) sum genericLength
-- ...

File: TypeScript/8 kyu - Calculate average.ts
// https://www.codewars.com/kata/calculate-average/train/typescript
// My solution
export function find_average(array) {
    return array ? array.reduce((a, b) =>  a + b) / array.length : 0;
    //return array ? array.reduce(function(a, b) { return a + b; }) / array.length : 0;
}

// ...
export function find_average(array) {
    return array.reduce((sum, value) => sum + value ) / array.length;
}
// ...
export function find_average(array) {
    // your code here
    return array.reduce((a,b)=>a+b,0)/array.length;
}
